lxc/list.go

```diff
@@ -512,41 +512,45 @@ func (c *cmdList) Run(cmd *cobra.Command, args []string) error {
 	if !nameFilter && needsData && d.HasExtension("container_full") {
 		// Using the GetInstancesFull shortcut
 		var cts []api.InstanceFull
+
+		serverFilters, clientFilters := d.GetServerSupportedFilters(filters, api.InstanceFull{})
 		if c.flagAllProjects {
-			cts, err = d.GetInstancesFullAllProjects(api.InstanceTypeAny)
+			cts, err = d.GetFilteredInstancesFullAllProjects(api.InstanceTypeAny, serverFilters)
 		} else {
-			cts, err = d.GetInstancesFull(api.InstanceTypeAny)
+			cts, err = d.GetFilteredInstancesFull(api.InstanceTypeAny, serverFilters)
 		}
 		if err != nil {
 			return err
 		}
 
-		return c.showInstances(cts, filters, columns)
+		return c.showInstances(cts, clientFilters, columns)
 	}
 
 	// Get the list of instances
 	var cts []api.Instance
 	var ctslist []api.Instance
+	serverFilters, clientFilters := d.GetServerSupportedFilters(filters, api.Instance{})
```

Comment:
Ah, I see why that function exists now :)

I'm still not sure it really fits with the `client` package though, so maybe easier to keep it in lxc/utils for now?